### ISSUE TITLE
508-defect-2: Extension input field does not have specific error messaging

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#53474]
### Point of contact

joshkimux

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

https://user-images.githubusercontent.com/14154792/218118927-b58bd5fa-e0dc-4e90-9352-b4fc14ba1503.mov

As seen in this video, it _appears_ that letters aren't accepted as inputs in the field which triggers a generic error. [To meet design system guidance](https://design.va.gov/patterns/ask-users-for/phone-numbers), the error should be:
- specific e.g. Please enter a X-digit extension number (with or without dashes)
- in-context to/attached directly to the field (as opposed to a separate alert)

### Reproduction steps

1. "On the edit home phone number page, tab to the extension field.
2. Turn on your screen reader.
3. Input "a" (which can't be accepted as it is a letter, not a number).
4. Click "save" and confirm a generic alert appears below the field.

### Proposed solution or next steps

An input specific validation should be provided if the user provides an invalid response within the extension field instead of a general alert. Focus should return to the field so the user can make an easy edit (if it's the first or only issue on the page).

### References, articles, or WCAG support

1. [Ask users for phone numbers](https://design.va.gov/patterns/ask-users-for/phone-numbers)
2. [Form error guidance](https://design.va.gov/components/form/#error-handling)

### Type of issue

- [X] Screenreader
- [ ] Keyboard
- [X] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by Mottie** [comment]
@saderagsdale Adam from the profile team will fix this problem for us!

**Comment by saderagsdale** [comment]
@Mottie Nice! 

@adamwhitlock1 Howdy! Thoughts on when this can be done?

**Comment by adamwhitlock1** [comment]
I will be pulling this work in shortly, and should have it completed by the end of my current sprint, which is 3/1

**Comment by adamwhitlock1** [comment]
@BerniXiongA6 and @Samara-Strauss Benefits team asked if I could take a look at this, and since I'm almost done with this sprints work, I will try and tackle this as well.

**Comment by BerniXiongA6** [comment]
Thanks for the heads up @adamwhitlock1 !

**Comment by adamwhitlock1** [comment]
@joshkimux I updated the phone extension to show a validation error when there are letters entered into the field, and have validated that the error prevents submission. So this should be fixed if you would like to review it, thanks!

**Comment by joshkimux** [comment]
Will do! Thanks so much @adamwhitlock1

**Comment by joshkimux** [comment]
https://user-images.githubusercontent.com/14154792/222455111-ff24b35e-0815-4822-8339-136d9ccde030.mov

Confirming this has been fixed! @adamwhitlock1 stellar as always 🤩
